src/ComplexType.php

```diff
@@ -136,59 +198,57 @@ protected function generateClass()
             } else {
                 $getterCode = '  return $this->' . $name . ';' . PHP_EOL;
             }
-            $getter = new PhpFunction('public', 'get' . ucfirst($name), '', $getterCode, $getterComment);
-            $accessors[] = $getter;
 
-            $setterComment = new PhpDocComment();
-            $setterComment->addParam(PhpDocElementFactory::getParam($type, $name, ''));
-            $setterComment->setReturn(PhpDocElementFactory::getReturn($this->phpNamespacedIdentifier, ''));
+            $getter->setBody($getterCode);
+            $this->class->addMethodFromGenerator($getter);
+
+            $setterParameter = new ParameterGenerator;
+            $setterParameter->setName($name);
+
+            if (isset($typeHint)) {
+                $setterParameter->setType($typeHint);
+
+                if ($member->getNullable()) {
+                    $setterParameter->setDefaultValue(
+                        new ValueGenerator(null, ValueGenerator::TYPE_NULL)
+                    );
+                }
+            }
```

Comment:
@kasperg: Completely refactored whole ```generateClass``` method. Should look more clear now. One small "sacrifice" - if constructor have no parameters there will be an empty dock block.